src/index.tsx

```diff
@@ -74,13 +75,21 @@ function getSelectedItem(widget: FileBrowser | null): Contents.IModel | null {
   return firstItem;
 }
 
-// Get only the file name, with no parent directories, of the currently selected file.
-function getSelectedFileName(widget: FileBrowser | null): string | null {
+// Get only the file base name, with no parent directories and no extension,
+// of the currently selected file.
+function getSelectedFileBaseName(widget: FileBrowser | null): string | null {
   const selectedItem = getSelectedItem(widget);
   if (selectedItem === null) {
     return null;
   }
-  return selectedItem.name;
+  const parts = selectedItem.name.split('.');
+  if (parts.length === 1) {
+    // no extension
+    return parts[0];
+  }
+
+  parts.splice(-1); // Remove the extension
+  return parts.join('.');
```

Comment:
Why not just always return `parts[0]`? This logic coerces `filename.tar.gz` => `filename.tar` rather than `filename`.

Reply:
Returning `parts[0]` would coerce `Cloud 3.0 release plan.ipynb` to `Cloud 3`, though

Reply:
Hmmm, alright, I see arguments can be made for both sides.